src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanCreator.java

```diff
@@ -114,6 +121,19 @@ private Map<String, String> getReplacements(ProgrammingLanguage programmingLangu
             String staticCodeAnalysisScript = createStaticCodeAnalysisScript(programmingLanguage, projectType);
             replacements.put(REPLACE_STATIC_CODE_ANALYSIS_SCRIPT, staticCodeAnalysisScript);
         }
+        // replace testwise coverage analysis placeholder
+        String testwiseCoverageAnalysisMavenProfile = "";
+        String testwiseCoverageAnalysisGradleTask = "";
+        String testwiseCoverageAnalysisScript = "";
+        if (isTestwiseCoverageAnalysisEnabled) {
+            testwiseCoverageAnalysisMavenProfile = "-Pcoverage";
+            testwiseCoverageAnalysisGradleTask = "tiaTests --run-all-tests";
+            testwiseCoverageAnalysisScript = createTestwiseCoverageAnalysisScript(projectType);
+        }
+        replacements.put(REPLACE_TESTWISE_COVERAGE_MAVEN_PROFILE, testwiseCoverageAnalysisMavenProfile);
+        replacements.put(REPLACE_TESTWISE_COVERAGE_GRADLE_TASK, testwiseCoverageAnalysisGradleTask);
+        replacements.put(REPLACE_TESTWISE_COVERAGE_SCRIPT, testwiseCoverageAnalysisScript);
```

Comment:
For the staticCodeAnalysis the replacement is inside the if, for the testwiseCoverage the replacement is outside the if statement. Could you unify that, or is there a reason for the difference?

Reply:
To give some context. For both SCA and non-SCA build plans, a different Jenkinsfile exists. Only the "SCA-file" contains the placeholder `#staticCodeAnalaysisScript`, the "non-SCA-file" does not contain this placeholder.
To cover all 4 options (SCA-Coverage, nSCA-Coverage, SCA-nCoverage, nSCA-nCoverage), I decided to always include the placeholder `#testwiseCoverageAnalysisReport` into the Jenkinsfile. In a follow-up-PR, I would unify this by always including the SCA-placeholder and therefore reducing the number of Jenkinsfiles to one (per project type).